src/mxcp/server/interfaces/server/mcp.py

```diff
@@ -389,6 +393,39 @@ def get_endpoint_counts(self) -> dict[str, int]:
             "total": tool_count + resource_count + prompt_count,
         }
 
+    def validate_all_endpoints(self) -> list[dict[str, str]]:
+        """Validate all endpoints and return any validation errors.
+
+        This method validates endpoints using the execution engine to check SQL syntax,
+        etc. Call this before run() to check for validation errors.
+
+        Returns:
+            List of validation errors as dicts with 'path' and 'error' keys.
```

Comment:
The function returns a `dict` with explicitly documented keys ("path" and "error").

Shouldn't that be a pydantic model?